core-primitives/stf-executor/src/executor.rs

```diff
@@ -24,45 +24,59 @@ use crate::{
 	BatchExecutionResult, ExecutedOperation,
 };
 use codec::{Decode, Encode};
+use ita_sgx_runtime::Runtime;
 use ita_stf::{
 	hash::{Hash, TrustedOperationOrHash},
 	stf_sgx::{shards_key_hash, storage_hashes_to_update_per_shard},
-	ParentchainHeader, ShardIdentifier, StateTypeDiff, Stf, TrustedGetterSigned, TrustedOperation,
+	Getter, ParentchainHeader, ShardIdentifier, Stf, TrustedCallSigned, TrustedGetterSigned,
+	TrustedOperation,
 };
 use itp_node_api::metadata::{pallet_teerex::TeerexCallIndexes, provider::AccessNodeMetadata};
 use itp_ocall_api::{EnclaveAttestationOCallApi, EnclaveOnChainOCallApi};
-use itp_sgx_externalities::SgxExternalitiesTrait;
+use itp_sgx_externalities::{SgxExternalitiesTrait, StateHash};
+use itp_stf_interface::{
+	parentchain_pallet::ParentchainPalletInterface, ExecuteCall, ExecuteGetter, StateCallInterface,
+	StateGetterInterface, UpdateState,
+};
 use itp_stf_state_handler::{handle_state::HandleState, query_shard_state::QueryShardState};
 use itp_time_utils::duration_now;
 use itp_types::{storage::StorageEntryVerified, OpaqueCall, H256};
 use log::*;
-use sp_runtime::{app_crypto::sp_core::blake2_256, traits::Header as HeaderTrait};
+use sp_runtime::traits::Header as HeaderTrait;
 use std::{
-	collections::BTreeMap, fmt::Debug, format, result::Result as StdResult, sync::Arc,
-	time::Duration, vec::Vec,
+	collections::BTreeMap, fmt::Debug, format, marker::PhantomData, result::Result as StdResult,
+	sync::Arc, time::Duration, vec::Vec,
 };
 
-pub struct StfExecutor<OCallApi, StateHandler, NodeMetadataRepository> {
+pub struct StfExecutor<OCallApi, StateHandler, NodeMetadataRepository, Stf, Call, Getter> {
```

Comment:
I believe the `Call` and `Getter` type parameters can be safely removed, they are not used? That should be an easy fix and I think it should be done now, to prevent adding unnecessary complexity to this type. 

I also noticed that now this executor is no longer limited to `std`. And along with that, the tests can be changed to `cargo` tests. But this might be what you meant in your last comment about cleaning up in a separate task?

Reply:
Yeah. I don't want to do this in this PR. I don't want to blow this up any further. Check out my proposed follow-up issues in the PR description:

> clean up of executor.rs and getter-executor. The executor (if still necessary) should now be available in std modus as well.

Can you agree to this?

Regarding the Call and Getter : You're right. I'll remove them.

Reply:
Yes, that works for me 👍

Reply:
#1035